### DISCUSSION TITLE
Intellij: Jbang Plugin on Mac M1 (Arm)

### DISCUSSION THREAD

**Discussion by jacquesvandermerwe** [jbangdev/jbang#1259] [General]
Good Morning

I'm running jbang (0.90.1) on a Mac M1 with the latest Intellij. The plugin however in Intellij raises the following error when trying to run the script and the IDE doesn't  recognize dependencies etc. Any advice?

`Unable to download JDK, unsupported Architecture: arm64` 

The balloon popup in Intellij raises the following:

`Unrecognized token 'Unable': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false') at [Source: (String)"Unable to download JDK, unsupported Architecture: arm64 "; line: 1, column: 7]`

**Comment by maxandersen** [comment]
Seeems like adoptium dont have a M1 compatible build of Java yet. You'll need to make Java Available in the path for jbang to find it until we have a M1 download Available.

**Reply by maxandersen** [reply]
hmm so https://github.com/adoptium/temurin-build/issues/1922 says M1 is available so this might be our download logic that is broken. 

@quintesse thoughts?

**Reply by quintesse** [reply]
Well the error message says it all, it's JBang itself complaining that it doesn't support "arm64".

So if we _can_ support it now because an Adoptium build for that architecture is available it might just be as simple as adding a new entry to this switch: https://github.com/jbangdev/jbang/blob/main/src/main/scripts/jbang#L90-L111

Can't really test that myself though.

**Reply by quintesse** [reply]
(We should also add it to this switch: https://github.com/jbangdev/jbang/blob/main/src/main/java/dev/jbang/util/Util.java#L417-L435 and add "arm" to it as well as that seems to be missing).

**Reply by maxandersen** [reply]
so I rented a M1 to see what happens - cloud ftw :)

and yes, adding arm64 works but only if choosing to get java 17 as for now java 11 builds are not available - see https://github.com/adoptium/adoptium/issues/96

**Reply by maxandersen** [reply]
what system would just return "arm" ?

**Reply by maxandersen** [reply]
got a fix for it in https://github.com/jbangdev/jbang/pull/1269 will release when i get some sleep.

**Reply by quintesse** [reply]
> what system would just return "arm" ?

No idea :-)
But it was added at some point so I guess (hope!) it was for a reason, right?

**Reply by jacquesvandermerwe** [reply]
Thanks @maxandersen for the prompt fix and response. Will the plugin update be released via the standard Intellij Plug-In mechanism

**Reply by maxandersen** [reply]
Ill need to Release jbang. Will try tonight. 

Then You just need to run jbang version --update.
